### PR TITLE
python3Packages.anyio: 3.3.2 -> 3.3.4

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change
https://anyio.readthedocs.io/en/stable/versionhistory.html

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 146367` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.hug</li>
<li>python38Packages.pytest-sanic</li>
<li>python39Packages.hug</li>
<li>python39Packages.pytest-sanic</li>
</ul>
</details>
<details>

  <summary>4 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>home-assistant: log was empty</li>
<li>python38Packages.httpx-socks: log was empty</li>
<li>python39Packages.httpx-socks: log was empty</li>
<li>wapiti: log was empty</li>
  </ul>
</details>
<details>

  <summary>35 packages failed to build and are new build failures:</summary>
  <ul>
<li>esphome: log was empty</li>
<li>platformio: log was empty</li>
<li>python38Packages.asyncclick: log was empty</li>
<li>python38Packages.authcaptureproxy: log was empty</li>
<li>python38Packages.cirq: log was empty</li>
<li>python38Packages.cirq-rigetti: log was empty</li>
<li>python38Packages.dependency-injector: log was empty</li>
<li>python38Packages.envoy-reader: log was empty</li>
<li>python38Packages.fastapi: log was empty</li>
<li>python38Packages.ha-philipsjs: log was empty</li>
<li>python38Packages.ismartgate: log was empty</li>
<li>python38Packages.json-logging: log was empty</li>
<li>python38Packages.jupyter-server-mathjax: log was empty</li>
<li>python38Packages.jupyter_server: log was empty</li>
<li>python38Packages.jupyterlab: log was empty</li>
<li>python38Packages.jupyterlab-git: log was empty</li>
<li>python38Packages.jupyterlab_server: log was empty</li>
<li>python38Packages.jupytext: log was empty</li>
<li>python38Packages.nbclassic: log was empty</li>
<li>python38Packages.nbdime: log was empty</li>
<li>python38Packages.pyquil: log was empty</li>
<li>python38Packages.python-kasa: log was empty</li>
<li>python38Packages.respx: log was empty</li>
<li>python38Packages.slowapi: log was empty</li>
<li>python38Packages.starlette: log was empty</li>
<li>python38Packages.teslajsonpy: log was empty</li>
<li>python39Packages.cirq: log was empty</li>
<li>python39Packages.jupyter-server-mathjax: log was empty</li>
<li>python39Packages.jupyter_server: log was empty</li>
<li>python39Packages.jupyterlab: log was empty</li>
<li>python39Packages.jupyterlab-git: log was empty</li>
<li>python39Packages.jupyterlab_server: log was empty</li>
<li>python39Packages.jupytext: log was empty</li>
<li>python39Packages.nbclassic: log was empty</li>
<li>python39Packages.nbdime: log was empty</li>
  </ul>
</details>
<details>
<summary>
98 packages built:
</summary>
<ul>
<li>apache-airflow (python39Packages.apache-airflow)</li>
<li>datasette (python39Packages.datasette)</li>
<li>dyndnsc</li>
<li>faraday-cli</li>
<li>gigalixir (python39Packages.gigalixir)</li>
<li>lexicon</li>
<li>mailman</li>
<li>odoo</li>
<li>pylode</li>
<li>python38Packages.amcrest</li>
<li>python38Packages.anyio</li>
<li>python38Packages.apache-airflow</li>
<li>python38Packages.asyncwhois</li>
<li>python38Packages.axis</li>
<li>python38Packages.datasette</li>
<li>python38Packages.datasette-template-sql</li>
<li>python38Packages.denonavr</li>
<li>python38Packages.elmax</li>
<li>python38Packages.entrance</li>
<li>python38Packages.entrance-with-router-features</li>
<li>python38Packages.falcon</li>
<li>python38Packages.gigalixir</li>
<li>python38Packages.httpcore</li>
<li>python38Packages.httpx</li>
<li>python38Packages.httpx-ntlm</li>
<li>python38Packages.iotawattpy</li>
<li>python38Packages.ipfshttpclient</li>
<li>python38Packages.ircrobots</li>
<li>python38Packages.mailman (python39Packages.mailman)</li>
<li>python38Packages.openapi-core</li>
<li>python38Packages.pylitterbot</li>
<li>python38Packages.pypoint</li>
<li>python38Packages.pyrmvtransport</li>
<li>python38Packages.pytest-httpx</li>
<li>python38Packages.qcs-api-client</li>
<li>python38Packages.rollbar</li>
<li>python38Packages.sanic</li>
<li>python38Packages.sanic-auth</li>
<li>python38Packages.sanic-testing</li>
<li>python38Packages.simple-rest-client</li>
<li>python38Packages.sunpy</li>
<li>python38Packages.total-connect-client</li>
<li>python38Packages.webargs</li>
<li>python38Packages.whodap</li>
<li>python38Packages.youtube-search-python</li>
<li>python38Packages.zeep</li>
<li>python39Packages.amcrest</li>
<li>python39Packages.anyio</li>
<li>python39Packages.asyncclick</li>
<li>python39Packages.asyncwhois</li>
<li>python39Packages.authcaptureproxy</li>
<li>python39Packages.axis</li>
<li>python39Packages.cirq-rigetti</li>
<li>python39Packages.datasette-template-sql</li>
<li>python39Packages.denonavr</li>
<li>python39Packages.dependency-injector</li>
<li>python39Packages.elmax</li>
<li>python39Packages.entrance</li>
<li>python39Packages.entrance-with-router-features</li>
<li>python39Packages.envoy-reader</li>
<li>python39Packages.falcon</li>
<li>python39Packages.fastapi</li>
<li>python39Packages.ha-philipsjs</li>
<li>python39Packages.httpcore</li>
<li>python39Packages.httpx</li>
<li>python39Packages.httpx-ntlm</li>
<li>python39Packages.iotawattpy</li>
<li>python39Packages.ipfshttpclient</li>
<li>python39Packages.ircrobots</li>
<li>python39Packages.ismartgate</li>
<li>python39Packages.json-logging</li>
<li>python39Packages.mailman-hyperkitty</li>
<li>python39Packages.openapi-core</li>
<li>python39Packages.pylitterbot</li>
<li>python39Packages.pypoint</li>
<li>python39Packages.pyquil</li>
<li>python39Packages.pyrmvtransport</li>
<li>python39Packages.pytest-httpx</li>
<li>python39Packages.python-kasa</li>
<li>python39Packages.qcs-api-client</li>
<li>python39Packages.respx</li>
<li>python39Packages.rollbar</li>
<li>python39Packages.sanic</li>
<li>python39Packages.sanic-auth</li>
<li>python39Packages.sanic-testing</li>
<li>python39Packages.simple-rest-client</li>
<li>python39Packages.slowapi</li>
<li>python39Packages.starlette</li>
<li>python39Packages.sunpy</li>
<li>python39Packages.teslajsonpy</li>
<li>python39Packages.total-connect-client</li>
<li>python39Packages.webargs</li>
<li>python39Packages.whodap</li>
<li>python39Packages.youtube-search-python</li>
<li>python39Packages.zeep</li>
<li>soco-cli</li>
<li>theharvester</li>
<li>ytmdl</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
odoo:
<details>

error: build log of '/nix/store/319jzwp6jdcnb5ydvbfw9zjky3mm8dll-odoo-15.0.20211029' is not available
When evaluating attribute ‘odoo’:
warning: unused-argument
Unused argument: `stdenv`.
Near pkgs/applications/finance/odoo/default.nix:1:3:

```
  |
1 | { stdenv
  |   ^
```
warning: unused-argument
Unused argument: `python3`.
Near pkgs/applications/finance/odoo/default.nix:4:3:

```
  |
4 | , python3
  |   ^
```
</details>
python38Packages.entrance:
<details>

error: build log of '/nix/store/irpz9zppfpc9q95q2qffvxlrwh53pv8p-python3.8-entrance-1.1.17' is not available
When evaluating attribute ‘python38Packages.entrance’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/entrance/default.nix:39:

```
   |
39 |     description = "A server framework for web apps with an Elm frontend";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.entrance-with-router-features:
<details>

error: build log of '/nix/store/4chkjaj4y40nr1j3iwzd24y9r38d071n-python3.8-entrance-1.1.17' is not available
When evaluating attribute ‘python38Packages.entrance-with-router-features’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/entrance/default.nix:39:

```
   |
39 |     description = "A server framework for web apps with an Elm frontend";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.respx:
<details>


warning: duplicate-check-inputs
Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.8-httpx-0.19.0

Near pkgs/development/python-modules/respx/default.nix:28:3:

```
   |
28 |   checkInputs = [
   |   ^
```

Near pkgs/development/python-modules/respx/default.nix:24:3:

```
   |
24 |   propagatedBuildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md
</details>
python38Packages.rollbar:
<details>

error: build log of '/nix/store/pqpj7xwhakj34abq38220w7pcmd61chw-python3.8-rollbar-0.16.2' is not available
When evaluating attribute ‘python38Packages.rollbar’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/rollbar/default.nix:45:5:

```
   |
45 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python39Packages.entrance:
<details>

error: build log of '/nix/store/c7ip39qscqbs5sy6i2zc0g1nbmq34ink-python3.9-entrance-1.1.17' is not available
When evaluating attribute ‘python39Packages.entrance’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/entrance/default.nix:39:

```
   |
39 |     description = "A server framework for web apps with an Elm frontend";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.entrance-with-router-features:
<details>

error: build log of '/nix/store/595wj8w7bzfb01fgx95zhy78igqp5c5v-python3.9-entrance-1.1.17' is not available
When evaluating attribute ‘python39Packages.entrance-with-router-features’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/entrance/default.nix:39:

```
   |
39 |     description = "A server framework for web apps with an Elm frontend";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.respx:
<details>

error: build log of '/nix/store/hlyjrkk5nwm4zwmh7wjib9j3gwjz3pba-python3.9-respx-0.17.1' is not available
When evaluating attribute ‘python39Packages.respx’:
warning: duplicate-check-inputs
Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.9-httpx-0.19.0

Near pkgs/development/python-modules/respx/default.nix:28:3:

```
   |
28 |   checkInputs = [
   |   ^
```

Near pkgs/development/python-modules/respx/default.nix:24:3:

```
   |
24 |   propagatedBuildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md
</details>
python39Packages.rollbar:
<details>

error: build log of '/nix/store/4alx2za32bj69qrabfhcwincszxk0vrx-python3.9-rollbar-0.16.2' is not available
When evaluating attribute ‘python39Packages.rollbar’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/rollbar/default.nix:45:5:

```
   |
45 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
</details>

**Review by risicle — APPROVED** [review]
`nixpkgs-review` reveals no new failures macos 10.15 & nixos x86_64

Package itself builds nixos aarch64 (emulated)
